### PR TITLE
redis: newer version and added TLS support

### PR DESCRIPTION
**Pull request by lpottier**
Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

### CODE REVIEW

File: var/spack/repos/builtin/packages/redis/package.py [hunk 1]
```diff
@@ -36,6 +50,25 @@ class Redis(MakefilePackage):
     version("4.0.12", sha256="6447259d2eed426a949c9c13f8fdb2d91fb66d9dc915dd50db13b87f46d93162")
     version("4.0.11", sha256="fc53e73ae7586bcdacb4b63875d1ff04f68c5474c1ddeda78f00e5ae2eed1bbb")
 
+    variant("tls", default=False, description="Builds with TLS support")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/redis/package.py]:
```suggestion
    variant("tls", default=False, when="@6:", description="Builds with TLS support")
```

File: var/spack/repos/builtin/packages/redis/package.py [hunk 2]
```diff
@@ -36,6 +50,25 @@ class Redis(MakefilePackage):
...
+    depends_on("openssl@1.1:", type=("build", "link"), when="+tls")
+    conflicts("@:6", when="+tls", msg="TLS support requires Redis >= 6.0.")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/redis/package.py]:
```suggestion
```
A note that this includes all the versions in the 6 major release. According to the comment I think you wanted `@:5`.

**Review by alalazo — CHANGES_REQUESTED** [review]
A minor request to use a conditional variant in place of a variant and a conflict.

**Comment by lpottier** [comment]
@alalazo Thanks for suggestions!

**Review by alalazo — APPROVED** [review]

**Pull request merged**